source/beamline/BioXAS/BioXASSSRLMonochromatorRegionControl.cpp

```diff
@@ -0,0 +1,846 @@
+{
+	if (!crystalChangeCCWLimitStatus_ && crystalChangeCCWLimitStatus_ != limitStatus) {
+		crystalChangeCCWLimitStatus_ = limitStatus;
+		addChildControl(crystalChangeCWLimitStatus_);
+
+		connect( crystalChangeCCWLimitStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+
+}
+
+void BioXASSSRLMonochromatorRegionControl::setRegionAStatusControl(AMControl *regionStatus)
+{
+	if (!regionAStatus_ && regionAStatus_ != regionStatus) {
+		regionAStatus_ = regionStatus;
+		addChildControl(regionAStatus_);
+
+		connect( regionAStatus_, SIGNAL(valueChanged(double)), this, SLOT(onRegionControlValueChanged()) );
+		connect( regionAStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setRegionBStatusControl(AMControl *regionStatus)
+{
+	if (!regionBStatus_ && regionBStatus_ != regionStatus) {
+		regionBStatus_ = regionStatus;
+		addChildControl(regionBStatus_);
+
+		connect( regionBStatus_, SIGNAL(valueChanged(double)), this, SLOT(onRegionControlValueChanged()) );
+		connect( regionBStatus_, SIGNAL(movingChanged(bool)), this, SIGNAL(movingChanged(bool)) );
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setValue(int newValue)
+{
+	if (value_ != newValue) {
+		value_ = newValue;
+		emit valueChanged(value_);
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setSetpoint(int newSetpoint)
+{
+	if (setpoint_ != newSetpoint) {
+		setpoint_ = newSetpoint;
+		emit setpointChanged(setpoint_);
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::setMoveInProgress(bool isMoving)
+{
+	if (moveInProgress_ != isMoving) {
+		moveInProgress_ = isMoving;
+		emit movingChanged(moveInProgress_);
+
+		if (moveInProgress_)
+			emit moveStarted();
+	}
+}
+
+void BioXASSSRLMonochromatorRegionControl::onActionStepChanged(int stepIndex)
+{
+	emit moveStepChanged(stepDescription(stepIndex), stepInstruction(stepIndex), stepNotes(stepIndex));
+}
+
+void BioXASSSRLMonochromatorRegionControl::onRegionControlValueChanged()
+{
+	BioXASSSRLMonochromator::Region::State newRegion = BioXASSSRLMonochromator::Region::None;
+
+	if (regionAStatus_ && regionAStatus_->isConnected() && regionBStatus_ && regionBStatus_->isConnected()) {
+
+		int regionAVal = (int)regionAStatus_->value();
+		int regionBVal = (int)regionBStatus_->value();
+
+		if (regionAVal == BioXASSSRLMonochromator::Region::NotIn && regionBVal == BioXASSSRLMonochromator::Region::In)
+			newRegion = BioXASSSRLMonochromator::Region::B;
+		else if (regionAVal == BioXASSSRLMonochromator::Region::In && regionBVal == BioXASSSRLMonochromator::Region::NotIn)
+			newRegion = BioXASSSRLMonochromator::Region::A;
+	}
+
+	setValue(newRegion);
+}
+
+void BioXASSSRLMonochromatorRegionControl::onRegionChangeStarted()
+{
+	setMoveInProgress(true);
+}
+
+void BioXASSSRLMonochromatorRegionControl::onRegionChangeCancelled()
+{
+	moveCleanup(sender());
```

Comment:
I agree that this should be changed. If it's alright, I'd like to take care of it in a separate issue.